optimum/onnxruntime/modeling_ort.py

```diff
@@ -138,18 +138,48 @@ def device(self) -> torch.device:
     def device(self, value: torch.device):
         self._device = value
 
-    def to(self, device: torch.device):
+    def _parse_device(self, device: Union[torch.device, str, int]):
```

Comment:
If you just want to map a device with its device id, I think it should be in utils instead of ort modeling. Have you checked `torch._C._nn._parse_to` for the parsing?

Reply:
Yes it's better indeed. I modified & retested according to your comment. Let me know if it is good.